mongo/src/main/java/com/redhat/lightblue/mongo/crud/IterateAndUpdate.java

```diff
@@ -160,14 +189,58 @@ public void update(CRUDOperationContext ctx,
                 }
                 docIndex++;
             }
+            // if we have any remaining items to update
+            if (numUpdating > 0) {
+                try {
+                    executeAndLogBulkErrors(bwo);
+                    numFailed += (int) docUpdateErrors.stream().map(e -> e.getIndex()).distinct().count();
+                } catch (Exception e) {
+                    LOGGER.warn("Update exception for documents for query: {}", query.toString());
+                }
+            }
         } finally {
             if (cursor != null) {
                 cursor.close();
             }
         }
-        response.setNumUpdated(numUpdated);
+        handleBulkWriteError(docUpdateErrors, docUpdateAttempts);
+
+        ctx.getDocumentsWithoutErrors().stream().forEach(doc -> {
+            ctx.getFactory().getInterceptors().callInterceptors(InterceptPoint.POST_CRUD_UPDATE_DOC, ctx, doc);
+        });
+
+        response.setNumUpdated(ctx.getDocumentsWithoutErrors().size());
```

Comment:
getDocumentsWithoutErrors is not numUpdated, it is exactly what it says: docsWithoutErrors, and it is an expensive operation. Instead, set a variable where you process errors.